cluster/src/main/java/org/apache/iotdb/cluster/ClusterIoTDB.java

```diff
@@ -637,11 +631,33 @@ public void disablePrintClientConnectionErrorStack() {
     printClientConnectionErrorStack = false;
   }
 
+  public SyncDataClient getSyncDataClient(Node node, int readOperationTimeoutMS) {
+    SyncDataClient dataClient =
+        (SyncDataClient) clientManager.borrowSyncClient(node, ClientCategory.DATA);
+    if (dataClient != null) {
+      dataClient.setTimeout(readOperationTimeoutMS);
+    }
+    return dataClient;
+  }
+
+  public AsyncDataClient getAsyncDataClient(Node node, int readOperationTimeoutMS) {
+    try {
+      AsyncDataClient dataClient =
+          (AsyncDataClient) clientManager.borrowAsyncClient(node, ClientCategory.DATA);
+      if (dataClient != null) {
+        dataClient.setTimeout(readOperationTimeoutMS);
+      }
+      return dataClient;
+    } catch (IOException e) {
+      logger.warn("error when get async client", e);
+    }
+    return null;
```

Comment:
Instead of returning a NULL and cause a NPE, we might need to throw an exception to the upper layer?